src/platform/wasm/Core.zig

```diff
@@ -0,0 +1,297 @@
+            const x = @intToFloat(f64, js.machEventShift());
+            const y = @intToFloat(f64, js.machEventShift());
+            self.last_cursor_position = .{
+                .x = x,
+                .y = y,
+            };
+            break :blk Event{
+                .mouse_motion = .{
+                    .pos = .{
+                        .x = x,
+                        .y = y,
+                    },
+                },
+            };
+        },
+        .mouse_press => Event{
+            .mouse_press = .{
+                .button = toMachButton(js.machEventShift()),
+                .pos = self.last_cursor_position,
+                .mods = self.last_key_mods,
+            },
+        },
+        .mouse_release => Event{
+            .mouse_release = .{
+                .button = toMachButton(js.machEventShift()),
+                .pos = self.last_cursor_position,
+                .mods = self.last_key_mods,
+            },
+        },
+        .mouse_scroll => Event{
+            .mouse_scroll = .{
+                .xoffset = @floatCast(f32, std.math.sign(js.machEventShiftFloat())),
+                .yoffset = @floatCast(f32, std.math.sign(js.machEventShiftFloat())),
+            },
+        },
+        .framebuffer_resize => blk: {
+            const width = @intCast(u32, js.machEventShift());
+            const height = @intCast(u32, js.machEventShift());
+            const pixel_ratio = @intCast(u32, js.machEventShift());
+            break :blk Event{
+                .framebuffer_resize = .{
+                    .width = width * pixel_ratio,
+                    .height = height * pixel_ratio,
+                },
+            };
+        },
+        .focus_gained => Event.focus_gained,
+        .focus_lost => Event.focus_lost,
+        else => null,
+    };
+}
+
+pub fn framebufferSize(self: *Core) Size {
+    return .{
+        .width = js.machCanvasFramebufferWidth(self.id),
+        .height = js.machCanvasFramebufferHeight(self.id),
+    };
+}
+
+pub fn setWaitTimeout(_: *Core, timeout: f64) void {
+    js.machSetWaitTimeout(timeout);
+}
+
+pub fn setTitle(self: *Core, title: [:0]const u8) void {
+    js.machCanvasSetTitle(self.id, title.ptr, title.len);
+}
+
+pub fn setDisplayMode(self: *Core, mode: DisplayMode, monitor: ?usize) void {
+    _ = monitor;
+    js.machCanvasSetDisplayMode(self.id, @enumToInt(mode));
+}
+
+pub fn displayMode(self: *Core) DisplayMode {
+    return @intToEnum(DisplayMode, js.machDisplayMode(self.id));
+}
+
+pub fn setBorder(self: *Core, value: bool) void {
+    _ = self;
+    _ = value;
+}
+
+pub fn border(self: *Core) bool {
+    _ = self;
+    return undefined;
+}
+
+pub fn setHeadless(self: *Core, value: bool) void {
+    _ = self;
+    _ = value;
+}
+
+pub fn headless(_: *Core) bool {
+    unreachable;
+}
+
+pub fn setVSync(self: *Core, mode: VSyncMode) void {
+    _ = self;
+    _ = mode;
+    unreachable;
+}
+
+pub fn vsync(_: *Core) VSyncMode {
+    unreachable;
```

Comment:
We should add a TODO here noting: https://github.com/gpuweb/gpuweb/issues/1224

But I think `unreachable` is the wrong behavior for no, we should just return the "vsync on" mode enum.